bots/heroes/dr_main.lua

```diff
@@ -0,0 +1,372 @@
+	local nValue = 0
+
+	if skills.abilQ:CanActivate() then
+		nValue = nValue + object.nContrpUp
+	end
+
+	if skills.abilW:CanActivate() then
+		nValue = nValue + object.nOpposingUp
+	end
+
+	if skills.abilR:CanActivate() then
+		nValue = nValue + object.nSpeedUp
+	end
+
+	if core.itemMorph and core.itemMorph:CanActivate() then
+		nValue = nValue + object.nMorphUp
+	end
+
+	if core.itemHFlower and core.itemHFlower:CanActivate() then
+		nValue = nValue + object.nHFlowerUp
+	end
+
+	return nValue
+end
+behaviorLib.CustomHarassUtility = CustomHarassUtilityFnOverride  
+
+-- @param botBrain: CBotBrain
+-- @return: none
+--
+local function HarassHeroExecuteOverride(botBrain)
+	local unitTarget = behaviorLib.heroTarget
+	if unitTarget == nil then
+		return object.harassExecuteOld(botBrain)
+	end
+    
+	local unitSelf = core.unitSelf
+	local vecMyPosition = unitSelf:GetPosition()
+--  local nAttackRange = core.GetAbsoluteAttackRangeToUnit(unitSelf, unitTarget)
+--  local nMyExtraRange = core.GetExtraRange(unitSelf)
+    
+	local vecTargetPosition = unitTarget:GetPosition()
+--  local nTargetExtraRange = core.GetExtraRange(unitTarget)
+	local nTargetDistanceSq = Vector3.Distance2DSq(vecMyPosition, vecTargetPosition)
+    
+	local nLastHarassUtility = behaviorLib.lastHarassUtil
+	local bCanSee = core.CanSeeUnit(botBrain, unitTarget)
+	local bActionTaken = false
+
+	if bCanSee then
+		local bTargetVuln = unitTarget:IsStunned() or unitTarget:IsImmobilized() or unitTarget:IsPerplexed()
+		core.FindItems()
+
+		if not bTargetVuln then
+			local bNeedRange = false
+			local nRangeNeeded = 0
+			if core.itemMorph ~= nil and core.itemMorph:CanActivate() then
+				local range = core.itemMorph:GetRange()
+				if nLastHarassUtility > botBrain.nMorphThreshold and nTargetDistanceSq < (range * range) then
+					bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, core.itemMorph, unitTarget)
+				end
+			end
+
+			if not bActionTaken then
+				-- try hell flower or opposing charges
+				if core.itemHFlower ~= nil and core.itemHFlower:CanActivate() 
+					and nLastHarassUtility > botBrain.nHFlowerThreshold
+				then
+					local range = core.itemHFlower:GetRange()
+					if nTargetDistanceSq < (range * range) then
+						bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, core.itemHFlower, unitTarget)
+					else
+						bNeedRange = true
+						nRangeNeeded = (range * range) - nTargetDistanceSq
+					end
+				elseif skills.abilW:CanActivate() and nLastHarassUtility > botBrain.nOpposingThreshold then
+					local range = skills.abilW:GetRange()
+					if nTargetDistanceSq < (range * range) then
+						bActionTaken = core.OrderAbilityEntity(botBrain, skills.abilW, unitTarget)
+
+						-- if we can cast Q, do it too
+						local qRange = skills.abilQ:GetRange()
+						if skills.abilQ:CanActivate()
+							and (unitSelf:GetLevel() >= 5 or nTargetDistanceSq < (qRange * qRange))
+						then
+							core.OrderAbility(botBrain, skills.abilQ)
```

Comment:
I don't think it's nessecary here, we already assigned to it before for the W, and that was the main purpose, Q is an extra.